File: 122.best-time-to-buy-and-sell-stock-ii.js
/*
 * @lc app=leetcode id=122 lang=javascript
 *
 * [122] Best Time to Buy and Sell Stock II
 */

// @lc code=start
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let diff = 0;
  if (prices.length > 0) {
    prices.reduce((acc, next) => {
      console.log(acc, next);
      if (next > acc) {
        diff += next - acc;
      }
      return next;
    });
  }
  return diff;
};
// @lc code=end
// 201/201 cases passed (64 ms)
// Your runtime beats 38.49 % of javascript submissions
// Your memory usage beats 90.48 % of javascript submissions (35.3 MB)

File: 122.best-time-to-buy-and-sell-stock-ii.java
/*
 * @lc app=leetcode id=122 lang=java
 *
 * [122] Best Time to Buy and Sell Stock II
 */

// @lc code=start
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        if (prices.length < 2) {
            return profit;
        }
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] > prices[i - 1]) {
                profit += prices[i] - prices[i - 1];
            }
        }
        return profit;
    }
}
// @lc code=end
// 201/201 cases passed (1 ms)
// Your runtime beats 90.56 % of java submissions
// Your memory usage beats 100 % of java submissions (36.9 MB)